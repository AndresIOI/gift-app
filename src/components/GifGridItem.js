import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem = ( {img, title} ) => {
  return (
    <div className='gif'>
      <img src={img} alt={title}/>
      <p>{title}</p>
    </div>
  )
}

export default GifGridItem;

GifGridItem.propTypes = {
  img: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}
