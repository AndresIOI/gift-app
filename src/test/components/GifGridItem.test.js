import React from 'react';
import { shallow } from "enzyme";

import  GifGridItem  from '../../components/GifGridItem'; 

describe('Pruebas en <GifGridItem />', () => {

  const url = 'https://google.com/goo.png';
  const title = 'Un Titulo';
  const wrapper = shallow(<GifGridItem img={url} title={title} />)

  test('debe de mostrar el componente correctamente', () => {
    expect( wrapper ).toMatchSnapshot();
  });

  test('debe de tener un parrafo con el title', () => {
    const p = wrapper.find('p');
    expect( p.text().trim() ).toBe( title );
  });

  test('debe de tener una imagen y un alt que sea igual al title', () => {
    const img = wrapper.find('img');
    expect( img.prop('src') ).toBe( url );
    expect( img.prop('alt') ).toBe( title );
  });

  test('debe de tener la clase gif', () => {
    const div = wrapper.find('div');
    expect( div.hasClass('gif') ).toEqual(true);
  });
});