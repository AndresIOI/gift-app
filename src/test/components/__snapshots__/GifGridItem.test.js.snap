// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de mostrar el componente correctamente 1`] = `
<div
  className="gif"
>
  <img
    alt="Un Titulo"
    src="https://google.com/goo.png"
  />
  <p>
    Un Titulo
  </p>
</div>
`;
